### PR TITLE
Decouple MCCY analytics and root folders

### PR DESCRIPTION
**Pull request by rafaelzaleski**
Closes 107-gh-Automattic/woopayments.

## Changes proposed in this Pull Request

This PR continues the MCCY migration to V2 by migrating the following:

- The multi-currency analytics folder.
- The multi-currency root folder.
- Specific components utilized within the context of multi-currency for these folders.

## Testing instructions

**Confirm Full Migration of `./client/multi-currency-analytics**
* Start by running the `npm run test:js` command and ensure all tests pass.
* Ensure all files in `./client/multi-currency-analytics` were moved to `./multi-currency/client/analytics`, properly disconnected from WooPayments, properly reconnected into MCCY, and interfaced with WooPayments through the interface later (e.g. `./multi-currency/interface/*`). 

**Confirm Full Migration of `./client/multi-currency**
* Start by running the `npm run test:js` command and ensure all tests pass.
* Ensure all files in `./client/multi-currency` were moved to `./multi-currency/client/`, properly disconnected from WooPayments, properly reconnected into MCCY, and interfaced with WooPayments through the interface later (e.g. `./multi-currency/interface/*`). 

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9326` or branch name `migrate/multi-currency-analytics` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=migrate/multi-currency-analytics) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 69be728ae93069ec07212e27eef74bdc066b8c48
- Build time: 2024-09-03 01:51:35 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: multi-currency/client/setup/tasks/add-currencies-task/index.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ import {
 
 // eslint-disable-next-line max-len
 // TODO: Migrate to 'multi-currency/settings/enabled-currencies-list/modal-checkbox-list'.
```
Comment (ricardo) [inline multi-currency/client/setup/tasks/add-currencies-task/index.js]: Should we remove these TODOs?

**Review by ricardo — APPROVED** [review]
Nice work. I couldn't find any issues in the code despite the large diff. 👍 

- ✅ Confirm Full Migration of `./client/multi-currency-analytics`.
- ✅ Confirm Full Migration of `./client/multi-currency`.
- ✅ Smoke tested multi currency and single currency settings.

⚠️ By opening the files from this PR I couldn't see the git blame, however the PR diff notes the files were moved/renamed, hopefully when this gets merged we will be able to see the git blame. Example:
- [multi-currency/client/data/actions.js](https://github.com/Automattic/woocommerce-payments/blame/e7b995a8f016466e7a24382e1198aa43303ef094/multi-currency/client/data/actions.js) – This was merged previously, I can see the git history.
- [multi-currency/client/analytics/index.js](https://github.com/Automattic/woocommerce-payments/blame/b1b60f8bba4f0caf93bbe8b8f739b549cc715a25/multi-currency/client/analytics/index.js) – From this PR, I can't see the git history.

File: multi-currency/client/setup/tasks/add-currencies-task/index.js (hunk 1 above)
Comment (lovo-h) [reply multi-currency/client/setup/tasks/add-currencies-task/index.js]: Good call. Fixed in [69be728](https://github.com/Automattic/woocommerce-payments/pull/9326/commits/69be728ae93069ec07212e27eef74bdc066b8c48).

**Review by ricardo — APPROVED** [review]

**Pull request merged**